app/vlinsert/loki/loki_json.go

```diff
@@ -153,16 +153,30 @@ func parseJSONRequest(data []byte, lmp insertutils.LogMessageProcessor, useDefau
 			}
 
 			// parse log message
-			msg, err := lineA[1].StringBytes()
-			if err != nil {
-				return fmt.Errorf("unexpected log message type for %q; want string", lineA[1])
+			msgV := lineA[1]
+			msgT := msgV.Type()
+			switch {
+			case msgT == fastjson.TypeString || len(msgFields) == 0:
```

Comment:
According to the Loki protocol format - `log line` must be a string. 

 https://github.com/grafana/loki/blob/ada4b7b8713385fbe9f5984a5a0aaaddf1a7b851/pkg/push/types.go#L26

Reply:
yes, indeed, closing PR